great_expectations/datasource/fluent/interfaces.py

```diff
@@ -548,14 +548,20 @@ def _add_asset(
 
         self.assets.append(asset)
 
-        self._save_context_project_config()
+        # if asset was added to a cloud FDS, _save_context_project_config will return FDS fetched from cloud,
+        # which will contain the new asset populated with an id
+        cloud_fds = self._save_context_project_config()
+        if cloud_fds:
+            # update self.assets to populate id field of new asset
+            self.assets = cloud_fds.assets
+            return self.get_asset(asset_name=asset.name)
```

Comment:
Why the additional retrieval if we have the asset from the fds?

Reply:
This is so that the asset returned by the method contains the new id added by cloud.